src/utils/dmg.c

```diff
@@ -345,29 +352,31 @@ pool_op_hdlr(int argc, char *argv[])
 		fprintf(stderr, "failed to parse service ranks\n");
 		return 2;
 	}
-	if (svc->rl_nr == 0) {
-		fprintf(stderr, "--svc mustn't be empty\n");
-		daos_rank_list_free(svc);
-		return 2;
-	}
 
-	targets = NULL;
-	if (tgt_str != NULL) {
-		targets = daos_rank_list_parse(tgt_str, ":");
-		if (targets != NULL && targets->rl_nr == 0) {
-			daos_rank_list_free(targets);
-			targets = NULL;
+	if (rank_str != NULL) {
+		ranks = daos_rank_list_parse(rank_str, ":");
+		if (ranks == NULL) {
+			fprintf(stderr, "failed to parse ranks\n");
+			return 2;
 		}
 	}
 
 	/* Check the targets for POOL_EXCLUDE, REPLICA_ADD & REPLICA_DEL. */
-	if (targets == NULL &&
+	if (ranks == NULL &&
```

Comment:
`ranks` is never really being used for replica_add or replica_del, right?

Reply:
Ah, it should use ranks in replica_add and del. Thanks.